GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -584,24 +619,33 @@ private void PostFetchJob(List<string> packIndexes)
 
                     using (ITracer activity = this.context.Tracer.StartActivity("TryWriteGitCommitGraph", EventLevel.Informational, Keywords.Telemetry, metadata: null))
                     {
-                        GitProcess process = new GitProcess(this.context.Enlistment);
-                        GitProcess.Result result = process.WriteCommitGraph(this.context.Enlistment.GitObjectsRoot, packIndexes);
+                        if (this.stopping)
+                        {
+                            this.context.Tracer.RelatedWarning(
+                                metadata: null,
+                                message: PostFetchTelemetryKey + ": Not launching 'git commit-graph' because the mount is stopping",
+                                keywords: Keywords.Telemetry);
+                            return;
+                        }
 
-                        if (result.HasErrors)
+                        GitProcess.Result result = this.postFetchGitProcess.WriteCommitGraph(this.context.Enlistment.GitObjectsRoot, packIndexes);
+
+                        if (!this.stopping && result.HasErrors)
                         {
+                            string message = PostFetchTelemetryKey
+                                + ": Failed to generate commit-graph for new packfiles"
+                                + (this.stopping ? " (expected because the process is stopping):" : ":")

```

Comment:
We already know `this.stopping` is false here (see line 633), so in this case we do not record a log event or telemetry on shutdown. Whatever the right pattern is, let's make sure the two code paths are consistent with each other.

Reply:
Oops. I had intended to remove the case from the `if` statement, but if we prefer to not write anything when `stopping`, then I'll leave it in the `if` and remove the tracing.